pegen/pegen.c

```diff
@@ -246,30 +247,24 @@ run_parser(struct tok_state* tok, void *(start_rule_func)(Parser *), int mode)
     void *res = (*start_rule_func)(p);
     if (res == NULL) {
         PyErr_Format(PyExc_SyntaxError, "error at mark %d, fill %d, size %d", p->mark, p->fill, p->size);
-        fail = 1;
         goto exit;
     }
 
+    if (mode == 1) {
+        result =  PyAST_mod2obj(res);
```

Comment:
FWIW I would love it if there was a mode that returned the compiled bytecode instead of the AST. That would allow a purer comparison with the current compiler (I believe that converting the internal AST to the public AST costs more than the rest of parsing or compilation). That's also something for another day...

Reply:
At some point it may be beneficial to use GitHub issues to make sure we do not forget to do these and to make sure we don't duplicate work.